leases/utils.py

```diff
@@ -1,12 +1,14 @@
 from datetime import date
 
 
-# If a lease object is being created before 10.6, then the dates are in the same year.
-# If the object is being created between those dates, then the start date is
-# the date of creation and end date is 14.9 of the same year.
-# If the object is being created after 14.9, then the dates are from next year.
-def calculate_berth_lease_start_date():
-    # Leases always start on 10.6 the earliest
+def calculate_season_start_date() -> date:
+    """Return the date when the summer season starts
+
+    If the current date is after the season end date,
+    it returns next year's default start date.
+
+    Leases always start on 10.6 the earliest
+    """
```

Comment:
I think this util should be more flexible in terms of the year. It should probably accept a date as parameter and take the year from that date, because in a similar case to the case where I found this bug - creating order for last year's lease (or for the next year's lease for that matter) - it would today's year and thus `is_full_season` would be `False` again 🤔

Reply:
I guess there should be a similar behavior for `calculate_season_end_date`? Also allowing to pass a date to base the year.

Reply:
yep